hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -477,4 +468,14 @@ public static MessageType readSchemaFromLogFile(FileSystem fs, Path path) throws
     }
     return null;
   }
+
+  public boolean hasOperationField() {
+    try {
+      Schema tableAvroSchema = getTableAvroSchemaFromDataFile();
```

Comment:
since this is called every time to retrieve the info, any performance concern on reading the timeline and the data files? can you help make a rough estimation on this aspect please? e.g. what kinds of scenarios may invoke this and how often it would happen

Reply:
> since this is called every time to retrieve the info, any performance concern on reading the timeline and the data files? can you help make a rough estimation on this aspect please? e.g. what kinds of scenarios may invoke this and how often it would happen

Thanks for your suggestion, I made a small adjustment. I will test the scene and time.